src/main/java/Duke.java

```diff
@@ -21,10 +22,52 @@ public static void exit() {
         printLine();
     }
 
-    public static void saveTask(Task task) {
+    public static void saveTask(String command) {
+        int start_idx, end_idx;
+        Task task;
+        String description, by, from, to;
+        String[] input = command.split(" ");
+
+        if (input[0].equalsIgnoreCase("todo")) {
+            start_idx = 1;
+            end_idx = input.length;
+            description = stringConverter(input, start_idx, end_idx);
+
+            task = new Todo(description);
+
+        } else if (input[0].equalsIgnoreCase("deadline")) {
+            start_idx = 1;
+            end_idx = Arrays.asList(input).indexOf("/by");
+            description = stringConverter(input, start_idx, end_idx);
+
+            start_idx = end_idx + 1;
+            end_idx = input.length;
+            by = stringConverter(input, start_idx, end_idx);
+
+            task = new Deadline(description, by);
+
+        } else {
+            start_idx = 1;
+            end_idx = Arrays.asList(input).indexOf("/from");
+            description = stringConverter(input, start_idx, end_idx);
+
+            start_idx = end_idx + 1;
```

Comment:
the number 1 could perhaps be defined as a constant named along the lines of 'zero-based indexing adjustment' so that it is very clear to those reading your code